File: ios/GildongmuKit/Sources/GildongmuKit/WhereAmIService.swift
import Foundation

/// "현재 위치 정위" 조회. GET /api/where-am-i?lat=&lng= envelope {data: WhereAmIData?}:
/// 키 없음→data:null(nil 반환, 死기능 아님), 네 조각 전부 비면 502(throw), 그 외 200+data.
public struct WhereAmIService: Sendable {
    let client: APIClient

    public init(client: APIClient) { self.client = client }

    public func locate(lat: Double, lng: Double) async throws -> WhereAmIData? {
        let response: WhereAmIResponse = try await client.get(
            "/api/where-am-i",
            query: [
                URLQueryItem(name: "lat", value: String(lat)),
                URLQueryItem(name: "lng", value: String(lng)),
            ])
        return response.data
    }
}

File: android/kit/src/main/kotlin/space/dodoplanet/gildongmu/kit/WhereAmIService.kt
package space.dodoplanet.gildongmu.kit

import space.dodoplanet.gildongmu.kit.models.WhereAmIData
import space.dodoplanet.gildongmu.kit.models.WhereAmIResponse

/**
 * "현재 위치 정위" 조회. Kit `WhereAmIService.swift` 미러. GET /api/where-am-i?lat=&lng= envelope {data: WhereAmIData?}:
 * 키 없음→data:null(null 반환, 죽은 기능 아님), 네 조각 전부 비면 502(throw), 그 외 200+data.
 */
class WhereAmIService(val client: APIClient) {
    suspend fun locate(lat: Double, lng: Double): WhereAmIData? =
        client.get<WhereAmIResponse>("/api/where-am-i", coordQuery(lat, lng)).data
}
